src/pudl_archiver/orchestrator.py

```diff
@@ -1,445 +1,58 @@
-
-        self.auto_publish = auto_publish
-        self.refresh_metadata = refresh_metadata
-
-        self.depositor = ZenodoDepositor(upload_key, publish_key, session, self.sandbox)
-        self.downloader = downloader
-
-        self.dry_run = dry_run
-
-        self.create_new = create_new
-        self.dataset_settings_path = dataset_settings_path
-        with Path.open(dataset_settings_path) as f:
-            self.dataset_settings = {
-                name: DatasetSettings(**dois)
-                for name, dois in yaml.safe_load(f).items()
-            }
-
-        self.changes: list[_DepositionChange] = []
-
-    async def _create_new_deposition(self) -> Deposition:
-        metadata = DepositionMetadata.from_data_source(self.data_source_id)
-        if not metadata.keywords:
-            raise AssertionError(
-                "New dataset is missing keywords and cannot be archived."
-            )
-        return await self.depositor.create_deposition(metadata)
-
-    async def _get_existing_deposition(
-        self, dataset_settings: dict[str, DatasetSettings], data_source_id: str
-    ) -> Deposition:
-        settings = dataset_settings[data_source_id]
-        doi = settings.sandbox_doi if self.sandbox else settings.production_doi
-        if not doi:
-            raise RuntimeError("Must pass a valid DOI if create_new is False")
-        return await self.depositor.get_deposition(doi)
-
-    async def run(self) -> RunSummary:
-        """Run the entire deposition update process.
-
-        1. Create pending deposition version to stage changes.
-        2. Download resources.
-        3. Update pending deposition version.
-        4. If there were updates, publish.
-        5. Update the dataset settings if this was a new deposition.
-
-        Returns:
-            RunSummary object.
-        """
-        self.changes = []
-        if self.create_new:
-            original = await self._create_new_deposition()
-            draft = original
-        else:
-            original = await self._get_existing_deposition(
-                self.dataset_settings, self.data_source_id
-            )
-            draft = await self.depositor.get_new_version(
-                original,
-                clobber=True,
-                data_source_id=self.data_source_id,
-                refresh_metadata=self.refresh_metadata,
-            )
-
-        resources = await self._download_then_upload_resources(draft, self.downloader)
-        for deletion in self._get_deletions(draft, resources):
-            await self._apply_change(draft, deletion)
-
-        draft = await self.depositor.get_deposition_by_id(draft.id_)
-        new_datapackage, old_datapackage = await self._update_datapackage(
-            original=original, draft=draft, resources=resources
-        )
-        run_summary = self._summarize_run(
-            old_datapackage, new_datapackage, resources, draft.links.html
+async def orchestrate_run(
```

Comment:
The orchestrator has been majorly simplified to just this top-level function, and most of the logic was moved to the `DraftDeposition` and the `ZenodoDepositorInterface` for zenodo specific stuff.